File: tests/fixtures/cpp/simple.cpp
// Simple C++ program for debugger testing
#include <iostream>
#include <map>
#include <string>

int add(int a, int b) {
    int result = a + b;
    return result;
}

int multiply(int a, int b) {
    int result = a * b;
    return result;
}

struct CalculationResult {
    int sum;
    int product;
};

CalculationResult calculate(int x, int y) {
    int sum_result = add(x, y);
    int product_result = multiply(x, y);
    return {sum_result, product_result};
}

int loop_example(int n) {
    int total = 0;
    for (int i = 0; i < n; i++) {
        total += i;
        std::cout << "Loop iteration " << i << ", total so far: " << total << std::endl;
    }
    return total;
}

int main() {
    std::cout << "Starting simple.cpp" << std::endl;

    // Test basic function calls
    int result = add(5, 3);
    std::cout << "add(5, 3) = " << result << std::endl;

    // Test nested function calls
    CalculationResult calc_result = calculate(4, 7);
    std::cout << "calculate(4, 7) = {sum: " << calc_result.sum
              << ", product: " << calc_result.product << "}" << std::endl;

    // Test loop
    int loop_result = loop_example(5);
    std::cout << "loop_example(5) = " << loop_result << std::endl;

    std::cout << "Finished simple.cpp" << std::endl;
    return 0;
}

File: tests/fixtures/swift/simple.swift
// Simple Swift program for debugger testing
import Foundation

func add(_ a: Int, _ b: Int) -> Int {
    let result = a + b
    return result
}

func multiply(_ a: Int, _ b: Int) -> Int {
    let result = a * b
    return result
}

struct CalculationResult {
    let sum: Int
    let product: Int
}

func calculate(_ x: Int, _ y: Int) -> CalculationResult {
    let sumResult = add(x, y)
    let productResult = multiply(x, y)
    return CalculationResult(sum: sumResult, product: productResult)
}

func loopExample(_ n: Int) -> Int {
    var total = 0
    for i in 0..<n {
        total += i
        print("Loop iteration \(i), total so far: \(total)")
    }
    return total
}

func main() {
    print("Starting simple.swift")

    // Test basic function calls
    let result = add(5, 3)
    print("add(5, 3) = \(result)")

    // Test nested function calls
    let calcResult = calculate(4, 7)
    print("calculate(4, 7) = {sum: \(calcResult.sum), product: \(calcResult.product)}")

    // Test loop
    let loopResult = loopExample(5)
    print("loopExample(5) = \(loopResult)")

    print("Finished simple.swift")
}

main()
